query-engine/connector-test-kit-rs/query-engine-tests/tests/writes/ids/uuid_create_graphql.rs

```diff
@@ -130,4 +130,62 @@ mod uuid_create_graphql {
 
         Ok(())
     }
+
+    fn schema_cuid_2() -> String {
+        let schema = indoc! {
+            r#"model Todo {
+              #id(id, String, @id, @default(cuid(2)))
+              title String
+            }"#
+        };
+
+        schema.to_owned()
+    }
+
+    // "Creating an item with an id field of model CUIDv2 and retrieving it" should "work"
+    #[connector_test(schema(schema_cuid_2))]
+    async fn create_cuid_v2_and_retrieve_it_should_work(runner: Runner) -> TestResult<()> {
+        let res = run_query_json!(
+            &runner,
+            r#"mutation {
+          createOneTodo(data: { title: "the title" }){
+            id
+          }
+        }"#
+        );
+
+        let cuid = match &res["data"]["createOneTodo"]["id"] {
+            serde_json::Value::String(str) => str,
+            _ => unreachable!(),
+        };
+
+        // Validate that this is a valid CUIDv2 value
+        {
+            assert!(cuid::is_cuid2(cuid.as_str()));
```

Comment:
this will have to be just
```suggestion
            assert!(cuid::is_cuid2(cuid));
```
after applying the other suggestion above